### PR TITLE
Test leftovers + generalized local writelock

### PR DESCRIPTION
**Pull request by plyhun**
## Abstract

Local writelock is moved into Core, to share with the alternative server implementations.

### General

* [ ] Added abstract that describes the change
* [ ] Added changelog entry to `/CHANGELOG.adoc`
* [ ] Ensured that the change is covered by tests
* [ ] Ensured that the change is documented in the docs

### On API Changes

* [ ] Checked if the changes are breaking or not
* [ ] Added GraphQL API if applicable
* [ ] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -181,36 +181,6 @@
 
 	<build>
 		<plugins>
-			<plugin>
```
Comment (plyhun) [inline core/pom.xml:184]: The test plugins are moved into the core tests, where they belong. The test plugins are baked into the core test resources, to be available for the alternative server implementations.

File: tests/tests-core/src/main/java/com/gentics/mesh/core/field/binary/BinaryFieldUploadEndpointParameterizedSynchTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.runners.Parameterized;
+
+import com.gentics.mesh.test.MeshTestSetting;
+
+@RunWith(Parameterized.class)
+@MeshTestSetting(testSize = FULL, startServer = true)
+public class BinaryFieldUploadEndpointParameterizedSynchTest extends AbstractBinaryFieldUploadEndpointParameterizedTest {
+
+	@Override
+	public boolean isSyncWrites() {
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/core/field/binary/BinaryFieldUploadEndpointParameterizedSynchTest.java:15]: The `syncWrites` change requires Mesh being restarted anew.

File: tests/tests-core/src/main/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java [hunk 3]
```diff
@@ -28,21 +30,6 @@
 @MeshTestSetting(testSize = FULL, startServer = false)
 public class DateListFieldMigrationTest extends AbstractFieldMigrationTest implements DateListFieldHelper {
 
-	/**
-	 * Cast a long to int if its possible. A long can be cast to int if it is smaller than Integer.MAX_VALUE.
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java:32]: Dead code.

File: tests/tests-core/src/main/java/com/gentics/mesh/plugin/AbstractPluginTest.java [hunk 4]
```diff
@@ -25,29 +27,31 @@
  */
 public class AbstractPluginTest extends AbstractMeshTest {
 
-	public static final String BASIC_PATH = "../../core/target/test-plugins/basic/target/basic-plugin-0.0.1-SNAPSHOT.jar";
+	public static final String PLUGINS_BASE = "/test-plugins/";
```
Comment (plyhun) [inline tests/tests-core/src/main/java/com/gentics/mesh/plugin/AbstractPluginTest.java:30]: Taken from now on from the jar resources, not from the filesystem.

**Review by npomaroli — APPROVED** [review]

**Pull request merged**